src/react/deckgl.js

```diff
@@ -100,12 +108,18 @@ export default class DeckGL extends React.Component {
     this._updateLayers(this.props);
 
     // Check if a mouse event has been specified and that at least one of the layers is pickable
-    const hasEvent = this.props.onLayerClick !== noop || this.props.onLayerHover !== noop;
+    const hasEvent =
```

Comment:
As discussed, we need to move this code out to a non-react dependent helper. There should just be a few calls in here that delegate to that helper.